external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -79,6 +82,7 @@ pub(super) struct DefnContext<'env, 'map> {
 struct Context<'env, 'map> {
     defn_context: DefnContext<'env, 'map>,
     address: Option<Address>,
+    warning_filters_table: Mutex<WarningFiltersTable>,
```

Comment:
I kept the `Arc` out of the `WarningFiltersTable` definition to allow for things like this. During construction, the table doesn't need the `Arc`, just this single Mutex. But once it is done, it needs to be in the `Arc` to prevent deep `clone`s--which would be unsafe